### PR TITLE
refactor: move to ruff + ruff-format

### PR DESCRIPTION
**Pull request by beckermr**
This PR moves us to ruff and ruff-format to avoid issues with flake8 pyproject support.

closes #108

### CODE REVIEW

**Comment by beckermr** [comment]
pre-commit.ci autofix

File: pyproject.toml [hunk 1]
```diff
@@ -1,27 +1,14 @@
-[tool.flake8]
+[tool.ruff.lint]
+select = ["E", "F", "I", "W"]
+ignore = ["C901", "E203", "W503", "E501"]
```
Comment (beckermr) [inline pyproject.toml:3]:
```suggestion
ignore = ["C901", "E203", "E501"]
```

**Comment by beckermr** [comment]
pre-commit.ci autofix

**Comment by beckermr** [comment]
pre-commit.ci autofix

**Review by ismael-mendoza — APPROVED** [review]
LGTM

**Pull request merged**